docs/NETWORK-ACLS.md

```diff
@@ -0,0 +1,289 @@
+where `type` is an enum, even if defined in the API as a string, while `value` is a string with semantics depending
+on the selected `type`. ACE can be configured with no matches (match-all), one match rule, or even multiple matches.
+However, multiple matches of the same type inside the same ACE are not allowed. Also, some of the combinations
+of different match types are not supported, see [limitations](#limitations).
+
+The currently supported ACE match types are:
+
+* `ip`: `value` should be an IP address of a remote endpoint. The match is satisfied for outbound and inbound flow
+  if the destination and the source IP address matches the given value, respectively.
+  Can be combined with any other match type to further narrow down the selection criteria.
+
+* `host`: `value` should be a domain name of a remote endpoint. It can be either a fully qualified, or a partially
+  qualified domain name (FQDN or PQDN). A packet is matched if it is destined to or originated from an IP address
+  that was obtained by a DNS query for that exact domain or any of its subdomains. For example, match of type `host`
+  with value `domain.com` will also apply to the endpoint `subdomain.domain.com`.
+  Can be combined with other match types except for `eidset` (see [limitations](#limitations)).
+
+* `eidset`: special match type for the overlay network. Matches IPs of all applications deployed in the same network
+  as well as all IPs with statically configured DNS entries (under the config field `NetworkInstanceConfig.Dns`).
+  For this type, `value` field is not used.
+  Can be combined with other match types except for `host` (see [limitations](#limitations)).
+
+* `protocol`: `value` should specify the protocol to match.
+  Protocol can be one of `tcp`, `udp`, `icmp`, or `all`, or it can be a numeric value, representing one of these
+  protocols or a different one. A protocol name from `/etc/protocols` is also allowed.
+  Protocol match can be combined with any other match type (often combined with port numbers).
+
+* `lport`: `value` should be an application *local* port number. For filtering actions, this is the source port for outbound
+  traffic and destination port for inbound traffic. For `PORTMAP` action, this represents application port as exposed
+  to the external network (i.e., if `<edge-node-ip>:2222` is mapped to `<app-ip>:22`, `lport` refers to `2222`).
+  `lport` can be combined with any other match type. It is actually required to combine `lport` and `protocol` inside
+  the same ACE.
+  In other words, port without protocol is not valid.
+
+* `fport`: `value` should be a remote endpoint port number (*foreign* port). Used for filtering actions,
+  but not for `PORTMAP` (do not confuse with `lport`, which is still used to represent the forwarded port - the forwarded
+  port is still considered as local).
+  `fport` can be combined with any other match type. It is actually required to combine `lport` with `protocol` inside
```

Comment:
Doesn't fport also require the protocol to be specified?

Reply:
Fixed - there was lport instead of fport in the description. Now it properly mentions that fport and protocol are required to be combined.